geode-core/src/main/java/org/apache/geode/cache/client/internal/PingOp.java

```diff
@@ -65,8 +71,9 @@ protected boolean needsUserId() {
     @Override
     protected void sendMessage(Connection cnx) throws Exception {
       getMessage().clearMessageHasSecurePartFlag();
-      this.startTime = System.currentTimeMillis();
-      getMessage().send(false);
+      getMessage().setNumberOfParts(1);
+      getMessage().addObjPart(serverID);
+      getMessage().send(true);
```

Comment:
Im afraid I dont have a reason to explain that change. I used a draft from @bschuchardt when implementing this, and I didnt realize about it. Bruce, could you help with this please?

Reply:
Ops that configure their message in constructors use AbstractOp.sendMessage().  This class already overrode that method so there's no reason to configure the message in the constructor.  That could be done though - is it a problem?

Maybe we don't have to clear the message in Message.send(boolean) but the default implementation, Message.send(), does this and since we've introduced a Part to this message I thought it prudent to clear it.

Reply:
No problem at all, I was just checking as I saw other messages execute the entire configuration logic (set number of parts, adding parts, etc.) directly in the constructor.

Reply:
thanks @bschuchardt